cockroachdb/sqlalchemy/transaction.py

```diff
@@ -65,6 +68,20 @@ def __exit__(self, typ, value, tb):
             savepoint_state.cockroach_restart = False
 
 
+def retry_exponential_backoff(retry_count: int) -> None:
+    """
+    This is a function for an exponential back-off whenever we encounter a retry error.
+    So we sleep for a bit before retrying,
+    and the sleep time increases for each failed transaction.
+
+    :param retry_count: The number for the current retry count
+    :return: None
+    """
+
+    sleep_secs = (2 ** retry_count) * 0.1 * (random() + 0.5)
```

Comment:
I think we would need to have a `max` here, otherwise the backoff could increase to something super large and unhelpful really quickly.

Also, using "jitter" has some benefits too. (See https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/) for example.)

So I think this formula should look more like:
```
random.uniform(0, min(max_backoff, 0.1 * (2 ** retry_count)))
```

Reply:
Hi @rafiss, thanks for the reviewing. I've made the changes you requested. Please take a look to see if there's more you want added to this PR.